### DISCUSSION TITLE
"magick compare" command: can I opt out of saving the output image?

### DISCUSSION THREAD

**Discussion by VioletGiraffe** [ImageMagick/ImageMagick#1990] [General] [answered]
According to [the docs](https://www.imagemagick.org/script/compare.php) and to my experiments, there is a mandatory third image argument to the `compare` command, e. g. `magick compare -metric rmse img.tiff reference.tiff output.jpg`. Can I opt out of it? I can see the use of the "difference map" image, but when there is no difference it's still written and contains something - I don't understand what. Also, writing this out image often takes longer than the comparison itself.

**Comment by urban-warrior** [comment, marked as answer]
Use `null:` as the output image.

**Comment by VioletGiraffe** [comment]
Thanks!
What's the meaning of the out image for the case when no difference is detected?
